ehrapy/tools/nlp/_medcat.py

```diff
@@ -156,213 +46,313 @@ def update_cat_config(self, concept_db_config: Config) -> None:
         """
         self.concept_db.config = concept_db_config
 
-    def extract_entities_text(self, text: str) -> Doc:
-        """Extracts entities for a provided text.
-
-        Args:
-            text: The text to extract entities from
-
-        Returns:
-            A spacy Doc instance. Extract the entities using. doc.ents
-        """
-        return self.cat(text)
-
-    def print_cui(self, doc: Doc, table: bool = False) -> None:
-        """Prints the concept unique identifier for all entities.
+    def set_filter_by_tui(self, tuis: list[str] | None = None):
+        """Restrict results of annotation step to certain tui's (type unique identifiers).
+        Note that this will change the MedCat object by updating the concept database config. In every annotation
+        process that will be run afterwards, entities are shown, only if they fall into the tui's type.
+        A full list of tui's can be found at: https://lhncbc.nlm.nih.gov/ii/tools/MetaMap/Docs/SemanticTypes_2018AB.txt
 
-        Args:
-            doc: A spacy tokens Doc.
-            table: Whether to print a Rich table.
-        """
-        if table:
-            # TODO IMPLEMENT ME
-            pass
-        else:
-            for entity in doc.ents:
-                # TODO make me pretty by ensuring that the lengths before the - token are aligned
-                print(f"[bold blue]{entity} - {entity._.cui}")
-
-    def print_semantic_type(self, doc: Doc, table: bool = False) -> None:
-        """Prints the semantic types for all entities.
+        As an exmaple:
+        Setting tuis=["T047", "T048"] will only annotate concepts (identified by a CUI (concept unique identifier)) in UMLS that are either diseases or
+        syndroms (T047) or mental/behavioural dysfunctions (T048). This is the default value.
 
         Args:
-            doc: A spacy tokens Doc.
-            table: Whether to print a Rich table.
-        """
-        # TODO implement Rich table
-        for ent in doc.ents:
-            print(ent, " - ", self.concept_db.cui2type_ids.get(ent._.cui))
-
-    def print_displacy(self, doc: Doc, style: Literal["deb", "ent"] = "ent") -> None:
-        """Prints a Doc with displacy
-
-        Args:
-            doc: A spacy tokens Doc.
-            style: The Displacy style to render
-        """
-        displacy.render(doc, style=style, jupyter=True)
-
-    def run_unsupervised_training(self, text: pd.Series, print_statistics: bool = False) -> None:
-        """Performs MedCAT unsupervised training on a provided text column.
+            tuis: list of TUI's
+        Returns:
 
-        Args:
-            text: Pandas Series of text to annotate.
-            print_statistics: Whether to print training statistics after training.
         """
-        print(f"[bold blue]Training using {len(text)} documents")
-        self.cat.train(text.values, progress_print=100)
+        if tuis is None:
```

Comment:
Do we really want this default filtering? Why is the default not exposed as a default parameter but rather hidden in the code?

I know that it takes ages if we don't do this filtering, but IMO the code must be more transparent then.

Reply:
Mutable types are not allowed (or at least not recommended) in python (as list would be). But we can discuss whether to remove this line completly of course. Might be more intuitive